chromium_src/chrome/browser/profiles/profile.cc

```diff
@@ -15,3 +17,10 @@ bool Profile::IsTorProfile() const {
   return false;
 #endif
 }
+
+bool IsGuestProfile(Profile* profile) {
+  DCHECK(profile);
+  return (profile->HasOffTheRecordProfile() &&
```

Comment:
circling back on this because the checks seems odd to me. Per the comment below shouldn't we be using `profile->IsGuestSession` instead of this method? I'm not sure that these OffTheRecordProfile checks are valid

Reply:
We call this check in `/browser/search_engines/search_engine_provider_service_factory.cc`'s function `InitializeSearchEngineProviderServiceIfNeeded`. The issue is that the stack for that call is like so:
<pre>
>	chrome.dll!`anonymous namespace'::InitializeSearchEngineProviderServiceIfNeeded(Profile * profile) Line 29	C++
 	chrome.dll!SearchEngineProviderServiceFactory::BuildServiceInstanceFor(content::BrowserContext * context) Line 70	C++
 	keyed_service_content.dll!BrowserContextKeyedServiceFactory::BuildServiceInstanceFor(void * context) Line 97	C++
 	keyed_service_core.dll!KeyedServiceFactory::GetServiceForContext(void * context, bool create) Line 77	C++
 	keyed_service_content.dll!BrowserContextKeyedServiceFactory::CreateServiceNow(void * context) Line 129	C++
 	keyed_service_core.dll!DependencyManager::CreateContextServices(void * context, bool is_testing_context) Line 70	C++
 	keyed_service_content.dll!BrowserContextDependencyManager::DoCreateBrowserContextServices(content::BrowserContext * context, bool is_testing_context) Line 47	C++
 	keyed_service_content.dll!BrowserContextDependencyManager::CreateBrowserContextServices(content::BrowserContext * context) Line 32	C++
 	chrome.dll!OffTheRecordProfileImpl::Init() Line 170	C++
 	chrome.dll!Profile::CreateOffTheRecordProfile() Line 679	C++
 	chrome.dll!ProfileImpl::GetOffTheRecordProfile() Line 883	C++
 	chrome.dll!ProfileManager::OnProfileCreated(Profile * profile, bool success, bool is_new_profile) Line 1258	C++
 	chrome.dll!BraveProfileManager::OnProfileCreated(Profile * profile, bool success, bool is_new_profile) Line 111	C++
</pre>

And `OffTheRecordProfileImpl::Init()` only sets it as a guest profile after it comes back from this call.  So at this point this profile would not return true for IsGuestSession. Also, we want to distinguish here between regular guest profile and otr guest profile, which IsGuestSession doesn't do.

Same for the second place where this function is used (DCHECK in `browser/search_engines/guest_window_search_engine_provider_service.cc`).

It seems pretty specific to the search engine provider, so maybe Profile is not the best place for it.

Reply:
addressed in 27492a7908b19ec3bd95aa16697b5aa6319018c8